### DISCUSSION TITLE
Service fails to start on Debian

### DISCUSSION THREAD

**Discussion by mohiazaddel** [rabbitmq/rabbitmq-server#2979] [Other]
i have this problem there is lot of discussion pages that say it will fix with adding your hostname to /etc/hosts but it doesnot work for me i tried installing with dpkg but nothing changed

```bash
● rabbitmq-server.service - RabbitMQ Messaging Server
   Loaded: loaded (/lib/systemd/system/rabbitmq-server.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Thu 2021-04-15 10:06:58 +0430; 4ms ago
  Process: 14133 ExecStart=/usr/sbin/rabbitmq-server (code=exited, status=1/FAILURE)
 Main PID: 14133 (code=exited, status=1/FAILURE)
dpkg: error processing package rabbitmq-server (--configure):
 installed rabbitmq-server package post-installation script subprocess returned error exit status 1
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for systemd (1:241-7~deb10u5+mx19) ...
Errors were encountered while processing:
 rabbitmq-server
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
We cannot comment on whether adding `localhost` to the hosts file will help. There is no context other than "the service has failed to start". Nodes [must be able to resolve their own hostname](https://www.rabbitmq.com/clustering.html#cluster-formation-requirements) but that's just one possible reason.

Please collect and share service and [RabbitMQ node logs](https://www.rabbitmq.com/logging.html) or we won't be able to help you.

**Comment by michaelklishin** [comment]
Two other crucially important pieces of information is how RabbitMQ was installed and what Erlang version is available to it. All we can tell from this output is that it is a reasonably modern Debian-based system. Modern RabbitMQ versions [require Erlang 22.3 or later](https://www.rabbitmq.com/which-erlang.html). A RabbitMQ 3.8.14 node won't start on Erlang 21.3. Our [Debian installation guide](https://www.rabbitmq.com/install-debian.html) explains how to provision Erlang 23.3 from Launchpad or Cloudsmith.

**Comment by michaelklishin** [comment]
Specifically `dpkg -i` also means you **must** [manually install all dependencies](https://www.rabbitmq.com/install-debian.html#manual-installation). `dpkg` does not manage dependencies, so either use the regular `apt` repo route or dependencies are on you.
